File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/utils/ProxyUtil.java
package tw.nekomimi.nekogram.utils;
import android.app.Activity;
import android.app.AlertDialog;
import android.content.Context;
import android.graphics.Bitmap;
import android.net.Uri;
public class ProxyUtil {
    public static void init() {}
    public static Activity getOwnerActivity(Context c) { return (c instanceof Activity) ? (Activity)c : null; }
    public static void showQrDialog(Context c, String text) {}
    public static AlertDialog showQrDialog(Context c, String text, String title) { return null; }
    public static AlertDialog showQrDialog(Activity a, String text) { return null; }
    public static String tryReadQR(Bitmap bitmap) { return null; }
    public static String tryReadQR(Activity activity, Uri uri) { return null; }
    public static void processDeepLink(Object activity, Uri uri, Object cb, Runnable fallback) {}
    public static void registerNetworkCallback() {}
    public static void showLinkAlert(Activity a, String url) {}
    public static boolean isIpv6Address(String addr) { return addr != null && addr.contains(":"); }
    public static boolean isVPNEnabled() { return false; }
    public static void importFromClipboard(Activity a) {}
}

File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/utils/ProxyUtil.kt
package tw.nekomimi.nekogram.utils

import android.app.Activity
import android.content.Context
import android.graphics.Bitmap
import org.telegram.ui.ActionBar.AlertDialog

object ProxyUtil {

    @JvmStatic
    fun isVPNEnabled(): Boolean = false

    @JvmStatic
    fun registerNetworkCallback() {}

    @JvmStatic
    fun getOwnerActivity(ctx: Context): Activity {
        error("ProxyUtil stub")
    }

    @JvmStatic
    @JvmOverloads
    fun showQrDialog(ctx: Context, text: String, icon: ((Int) -> Bitmap)? = null): AlertDialog {
        return AlertDialog.Builder(ctx).create()
    }

    @JvmStatic
    fun createQRCode(text: String, size: Int = 768, icon: ((Int) -> Bitmap)? = null): Bitmap {
        return Bitmap.createBitmap(size, size, Bitmap.Config.ARGB_8888)
    }

    @JvmStatic
    fun tryReadQR(ctx: Activity, bitmap: Bitmap) {}

    @JvmStatic
    @JvmOverloads
    fun showLinkAlert(ctx: Activity, text: String, tryInternal: Boolean = true) {}

    @JvmStatic
    fun importFromClipboard(ctx: Activity) {}

    @JvmStatic
    fun isIpv6Address(value: String): Boolean = false
}
